### PR TITLE
Issue 27 readme

### PR DESCRIPTION
**Pull request by aureamunoz**
I will embedd the video when it will be ready, but you can already take a look and give some feed-backs :-)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,17 @@
 # hal
-Easily create and manage Kubernetes applications using Dekorate and the Halkyon operator, made with ❤️ by the Snowdrop team.
+
+## Overview
+`hal` is a CLI tool for developers to create and manage Spring Boot applications. Simplify the deployment of applications on OpenShift and Kubernetes using Dekorate and Halkyon Component Operator. Made with ❤️ by the Snowdrop team.
```
Comment (cmoulliard) [inline README.md]: This is true for the moment as only spring boot project is scaffolded but not at all regarding to what `hal` proposes globally. I propose to rephrase to `Hal is a CLI tool for developers to simplify the deployment of applications such as Spring Boot on OpenShift and Kubernetes using Dekorate and Halkyon Component Operator. Made with ❤️ by the Snowdrop team.`

File: README.md [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
 [![CircleCI](https://circleci.com/gh/halkyonio/hal.svg?style=svg)](https://circleci.com/gh/halkyonio/hal)
 
+## Key features
+`hal` is capable of doing the following tasks
+- Scaffold Spring Boot applications
+- Deploy Spring Boot applications as components
```
Comment (cmoulliard) [inline README.md]: To be renamed to `-Deploy Microservices application as Components`

File: README.md [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+- Switch the component between `dev` and `build` modes
```
Comment (cmoulliard) [inline README.md]: To be renamed to `Switch the deployment from the dev to build mode

File: README.md [hunk 4]
```diff
@@ -13,3 +22,58 @@ Easily create and manage Kubernetes applications using Dekorate and the Halkyon
...
+## Deploying a component using `hal`
+After installing `hal`, the following steps allows to create and deploy a project to a cluster.
+**Note**: this assumes that you are connected to a Halkyon-enabled OpenShift/Kubernetes cluster.
+
+###1 Scaffold the Spring Boot applications 
+
+ - Create a development folder on your laptop
+`mkdir haldemo && cd haldemo`
+
+Create a new project using the REST HTTP `rest` template proposed by the scaffolding tool:
```
Comment (cmoulliard) [inline README.md]: --> "using the HTTP `rest` template...."

File: README.md [hunk 5]
```diff
@@ -13,3 +22,58 @@ Easily create and manage Kubernetes applications using Dekorate and the Halkyon
...
+    -s 2.1.6.RELEASE \
+    -t rest \
+    -v 1.0.0-SNAPSHOT \
+    --supported=false \
+    hello-world
+```
+
+###2 Deploy the Component
+
+A component represents a micro-service, i.e. part of an application to be deployed. The Component custom resource provides a simpler to fathom abstraction over what's actually required at the Kubernetes level to deploy and optionally expose the micro-service outside of the cluster. In fact, when a component is deployed to a [Halkyon](https://github.com/halkyonio)-enabled cluster, the [Halkyon operator](https://github.com/halkyonio/operator) will create these OpenShift/Kubernetes resources.
```
Comment (cmoulliard) [inline README.md]: If you don't enumerate the k8s resources created, then please remove the test at the end of this paragrapha -> "will create these OpenShift/Kubernetes resources."

File: README.md [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
+## Overview
+Hal is a CLI tool for developers to simplify the deployment of applications such as Spring Boot on OpenShift and Kubernetes using Dekorate and Halkyon Component Operator. Made with ❤️ by the Snowdrop team.
 
 [![CircleCI](https://circleci.com/gh/halkyonio/hal.svg?style=svg)](https://circleci.com/gh/halkyonio/hal)
 
+## Key features
+`hal` is capable of doing the following tasks
+- Scaffold Spring Boot applications
+- Deploy Microservices applications as Components
+- Switch the component from the `dev` to the `build` mode
```
Comment (cmoulliard) [inline README.md]: No -> Should be: Switch the DeploymentMode of the Component from `dev` to `build`

File: README.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ Hal is a CLI tool for developers to simplify the deployment of applications such
 `hal` is capable of doing the following tasks
 - Scaffold Spring Boot applications
 - Deploy Microservices applications as Components
-- Switch the component from the `dev` to the `build` mode
+- Switch the component from `dev` to `build` mode
```
Comment (cmoulliard) [inline README.md]: --> "Switch the Deployment mode of the Component from `Dev` to `Build`"

File: README.md [hunk 8]
```diff
@@ -1,8 +1,29 @@
...
+- [Demonstration](#demonstration)
+- [Additional documentation](additional-documentation)
+
+## Overview
+Hal is a CLI tool for developers to simplify the deployment of applications such as Spring Boot on OpenShift and Kubernetes using Dekorate and Halkyon Component Operator. Made with ❤️ by the Snowdrop team.
 
 [![CircleCI](https://circleci.com/gh/halkyonio/hal.svg?style=svg)](https://circleci.com/gh/halkyonio/hal)
 
+## Key features
+`hal` is capable of doing the following tasks
```
Comment (metacosm) [inline README.md]: You should point to the Halkyon readme / doc to explain the context first as otherwise these features might not make any sense.

Comment (cmoulliard) [reply README.md]: I agree with @metacosm

Comment (aureamunoz) [reply README.md]: Done, WDYT? @cmoulliard @metacosm

**Comment by cmoulliard** [comment]
LGTM

File: README.md [hunk 9]
```diff
@@ -13,3 +34,56 @@ Easily create and manage Kubernetes applications using Dekorate and the Halkyon
...
+## Deploying a component using `hal`
+After installing `hal`, the following steps allows to create and deploy a project to a cluster.
+**Note**: this assumes that you are connected to a Halkyon-enabled OpenShift/Kubernetes cluster.
+
+### 1 Scaffold the Spring Boot applications 
+
+ - Create a development folder on your laptop
+`mkdir haldemo && cd haldemo`
+
+Create a new project using the HTTP `rest` template proposed by the scaffolding tool:
```
Comment (metacosm) [inline README.md]: Missing `-` to continue the list

File: README.md [hunk 10]
```diff
@@ -13,3 +34,56 @@ Easily create and manage Kubernetes applications using Dekorate and the Halkyon
...
+    -v 1.0.0-SNAPSHOT \
+    --supported=false \
+    hello-world
+```
+
+### 2 Deploy the Component
+
+A component represents a micro-service, i.e. part of an application to be deployed. The Component custom resource provides a simpler to fathom abstraction over what's actually required at the Kubernetes level to deploy and optionally expose the micro-service outside of the cluster. In fact, when a component is deployed to a [Halkyon](https://github.com/halkyonio)-enabled cluster, the [Halkyon operator](https://github.com/halkyonio/operator) will create these OpenShift/Kubernetes resources such as `Deployment`, `Service`, `PersistentVolumeClaim`, `Ingress` or `Route` on OpenShift if the component is exposed.
+
+`hal component push -c hello-world`
```
Comment (metacosm) [inline README.md]: You're missing a the build step first: `mvn -f hello-world`

File: README.md [hunk 11]
```diff
@@ -13,3 +34,56 @@ Easily create and manage Kubernetes applications using Dekorate and the Halkyon
...
+`oc get components`
+
+```
+NAME               RUNTIME       VERSION         AGE       MODE      STATUS    MESSAGE   REVISION
+hello-world        spring-boot   2.1.6.RELEASE   7m17s     dev       Ready     Ready     6aadfc1a982fcd68
+```
+
+### 3 Connect to the rest services
+
+Try the rest service to see if it works. To do so, get the route address of the microservice using this command `oc get routes/hello-world --template={{.spec.host}}`
```
Comment (metacosm) [inline README.md:85]: This assumes that you deploy on OpenShift. We should also add the example for plain Kubernetes.

File: pkg/hal/cli/component/push.go [hunk 12]
```diff
@@ -28,8 +28,8 @@ type pushOptions struct {
 }
 
 var (
-	pushExample = ktemplates.Examples(`  # Deploy the components client-sb, backend-sb
-  %[1]s -c client-sb,backend-sb`)
+	pushExample = ktemplates.Examples(`  # Deploy the 'foo' component
```
Comment (metacosm) [inline pkg/hal/cli/component/push.go]: Why is this change in this PR?

**Review by metacosm — CHANGES_REQUESTED** [review]

File: README.md (hunk 10 above)
Comment (aureamunoz) [reply README.md]: Yes, I realised, I've already added it.

File: pkg/hal/cli/component/push.go (hunk 12 above)
Comment (aureamunoz) [reply pkg/hal/cli/component/push.go]: Oups, I have slipt in this commit. I'll revert it and open another PR

**Review by metacosm — APPROVED** [review]

**Pull request merged**